src/test/kotlin/note/application/NoteUpdaterTest.kt

```diff
@@ -0,0 +1,65 @@
+package note.application
+
+import note.domain.*
+import note.domain.exceptions.NonExistentNoteException
+import note.mothers.IdentifierMother
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.Mockito
+
+class ReminderUpdaterTest {
+    private lateinit var reminderUpdater: ReminderUpdater
+    private lateinit var generator: NoteIdentifierGenerator
+    private lateinit var repository: NoteRepository
+
+    private val newTitle = "title"
+    private val newDescription = "description"
+    private val identifier = IdentifierMother.getValidIdentifier()
+
+    @BeforeEach
+    fun setUp() {
+        repository = Mockito.mock(NoteRepository::class.java)
+        generator = Mockito.mock(NoteIdentifierGenerator::class.java)
+        reminderUpdater = ReminderUpdater(repository, generator)
+    }
+
+    @Test
+    fun `Invalid identifier throws NonExistentReminderException and doesn't call save`() {
+        // Initialize
+        val identifier = IdentifierMother.getValidIdentifier()
+        Mockito.`when`(repository.search(identifier)).thenReturn(null)
+
+        // Execute and Assert
+        assertThrows<NonExistentNoteException> {
+            reminderUpdater.update(identifier, newTitle, newDescription)
+        }
+
+        // Assert number of calls
+        assertNumberOfCalls(1, 0, 0)
+    }
+
+    @Test
+    fun `Existing note gets deleted and saved`() {
+        // Initialize
+        Mockito.`when`(repository.search(identifier)).thenReturn(Note(identifier, Title("title"), Description("description")))
+        Mockito.`when`(generator.generate()).thenReturn(identifier)
+
+        // Execute and Assert
+        reminderUpdater.update(identifier, newTitle, newDescription)
+
+        // Assert number of calls
+        assertNumberOfCalls(1, 1, 1)
```

Comment:
Same as before only check if the new one is added and the old one deleted, no need to assert if the oldOne was searched.